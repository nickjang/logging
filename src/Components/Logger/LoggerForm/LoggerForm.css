.logger-form {
  width: fit-content;
  margin: auto;
}