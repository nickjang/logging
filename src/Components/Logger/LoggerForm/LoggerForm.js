import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ProjectSelect from '../ProjectSelect/ProjectSelect';
import LogButton from '../LogButton/LogButton';
import './LoggerForm.css';

class LogForm extends Component {
  render() {
    return (
      <form action='' id='project-form'>
        <ProjectSelect updateProject={this.props.updateProject} />
        <LogButton />
      </form>
    );
  }
}

ProjectSelect.defaultProps = {
  updateProject: ()=>{}
}

ProjectSelect.propTypes = {
  updateProject: PropTypes.func.isRequired
}

export default LogForm;