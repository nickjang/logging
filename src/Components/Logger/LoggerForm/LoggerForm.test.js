import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoggerForm from './LoggerForm';

describe(`LoggerForm component`, () => {
  it('renders LoggerForm by default', () => {
    const wrapper = shallow(<LoggerForm />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})