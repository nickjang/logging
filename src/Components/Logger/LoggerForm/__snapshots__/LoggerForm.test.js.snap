// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggerForm component renders LoggerForm by default 1`] = `
<form
  action=""
  className="logger-form group-col"
  id="project-form"
>
  <ProjectSelect />
  <LogButton />
</form>
`;
